### PR TITLE
Schemas

### PR DESCRIPTION
**Pull request by 211458-LakkeRajashekar**

### CODE REVIEW

File: src/components/ComposePage/TextArea/textArea.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import React from "react";
 import styles from "./textArea.module.css";
 
 const TextAreaComponent = (props) => {
+  console.log(props,"ppppppp")
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/TextArea/textArea.tsx]: remove console.log

File: src/pages/compose/index.tsx [hunk 2]
```diff
@@ -104,7 +105,11 @@ const Compose = () => {
     dispatch(clearLastIndex());
     setName("");
     setDescription("");
-    showSuccessToast("Deleted Successfully")
+    dispatch(deleteProjectInfoAction(id));
+    // dispatch(deleteProjectGroupsInfoAction(5));
+    showSuccessToast("Deleted Successfully");
```
Comment (sourcecoursepoc) [inline src/pages/compose/index.tsx]: display success message based on API response

File: src/redux/actions/fetchProjectAction.tsx [hunk 3]
```diff
@@ -16,43 +16,97 @@ import {
...
+export const deleteProjectInfoActionFailure = (error: any): DeleteProjectInfoActionFailure => {
+  return {
+    type: DELETE_PROJECTS_INFO_ACTION_FAILURE,
+    payload: { error },
+  };
+};
+
+//delete project groups
+
+export const deleteProjectGroupsInfoAction = (id: any): DeleteProjectGroupsInfoAction => {
```
Comment (sourcecoursepoc) [inline src/redux/actions/fetchProjectAction.tsx]: Is it required ?

File: src/redux/actions/projectActionTypes.tsx [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
 export const FETCH_ALLPROJECTS_SUCCESS = "FETCH_ALLPROJECTS_SUCCESS";
-export const FETCH_ALLPROJECTS_FAILURE = "FETCH_ALLPROJECTS_FAILURE";
\ No newline at end of file
+export const FETCH_ALLPROJECTS_FAILURE = "FETCH_ALLPROJECTS_FAILURE";
+
+export const DELETE_PROJECTS_INFO_ACTION = " DELETE_PROJECTS_INFO_ACTION";
+export const DELETE_PROJECTS_INFO_ACTION_SUCCESS = "DELETE_PROJECTS_INFO_ACTION_SUCCESS";
+export const DELETE_PROJECTS_INFO_ACTION_FAILURE = "DELETE_PROJECTS_INFO_ACTION_FAILURE";
+
+export const DELETE_PROJECTGROUPS_INFO_ACTION = " DELETE_PROJECTGROUPS_INFO_ACTION";
```
Comment (sourcecoursepoc) [inline src/redux/actions/projectActionTypes.tsx]: DELETE_PROJECTGROUPS Is it required ?

File: src/redux/sagas/projectSaga.tsx [hunk 5]
```diff
@@ -31,12 +32,68 @@ function* fetchProjectSaga():any {
...
+  axios.delete("http://localhost:8080/sourcecourse/project/" + requestParams);
+
+/*
+  Worker Saga: Fired on FETCH_SCHEMA_REQUEST action
+*/
+function* deleteProjectSaga(requestParams: DeleteProjectInfoAction) {
+
+  try {
+    const response = yield call(() => deleteProject(requestParams.payload));
+    console.log(response,requestParams,"rrrrrrrr")
```
Comment (sourcecoursepoc) [inline src/redux/sagas/projectSaga.tsx]: remove console.log

**Review by sourcecoursepoc — CHANGES_REQUESTED** [review]

File: src/components/ComposePage/buttons/buttonComponent.tsx [hunk 6]
```diff
@@ -52,7 +52,7 @@ const ButtonComponent = ({
 
 /*   const handleDeleteModalOk = () => {
     setDeleteModalVisible(false);
-    dispatch(clearLastIndex());
+    // dispatch(clearLastIndex());
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/buttons/buttonComponent.tsx]: Any purpose in keeping commented lines? Please remove

File: src/components/homeLeftArea/homeLeftArea.tsx [hunk 7]
```diff
@@ -13,7 +13,7 @@ import { getProjectsSelector } from "@/redux/selector";
 const HomeLeftArea: React.FunctionComponent = () => {
   const dispatch = useDispatch();
   const projectListData = useSelector(getProjectsSelector);
-  const projectArray = projectListData[0]?.projects;
+  const projectArray = projectListData[0] ?.projects;
```
Comment (sourcecoursepoc) [inline src/components/homeLeftArea/homeLeftArea.tsx:17]: Why there is a space here between projectListData[0] and  ?.projects

File: src/components/homeLeftArea/homeLeftArea.tsx [hunk 8]
```diff
@@ -46,28 +46,28 @@ const HomeLeftArea: React.FunctionComponent = () => {
...
-              <Row className={styles.contentStyle} key={item.projectName}>
-                <ProjectContent
-                  heading={item.projectName}
-                  projectDescription={item.description}
-                  uid={item.uid}
-                ></ProjectContent>
-              </Row>
-            ))
-          : []}
+          ? parentArray ?.map((item) => (
```
Comment (sourcecoursepoc) [inline src/components/homeLeftArea/homeLeftArea.tsx:50]: Why space

File: src/pages/compose/index.tsx [hunk 9]
```diff
@@ -60,7 +61,7 @@ const Compose = () => {
 
   const handleSaveProjectInfo = async () => { 
     setSaveModalVisible(false);
-    showSuccessToast("Saved Successfully")
+    showSuccessToast("Project Saved Successfully")
```
Comment (sourcecoursepoc) [inline src/pages/compose/index.tsx]: Move hardcoded values to a constant file

File: src/pages/compose/index.tsx [hunk 10]
```diff
@@ -104,7 +105,11 @@ const Compose = () => {
     dispatch(clearLastIndex());
     setName("");
     setDescription("");
-    showSuccessToast("Deleted Successfully")
+    dispatch(deleteProjectInfoAction(id));
+    // dispatch(deleteProjectGroupsInfoAction(5));
```
Comment (sourcecoursepoc) [inline src/pages/compose/index.tsx]: Need to be removed

File: src/pages/schemas/content.tsx [hunk 11]
```diff
@@ -51,7 +51,7 @@ export default function SchemaContent() {
             await dispatch(postColumnTagsAndDescriptionInfoAction(selectedUid, tags, description));
         }
         showSuccessToast("saved successfully");
-        dispatch(fetchDataBaseInfoAction());
+        // dispatch(fetchDataBaseInfoAction());
```
Comment (sourcecoursepoc) [inline src/pages/schemas/content.tsx]: Again make sure no commented lines should be included for merging in main branch

File: src/redux/actions/types.tsx [hunk 12]
```diff
@@ -160,138 +161,162 @@ export type SchemaActions =
...
+export interface FetchProjectRequest {
+  type: typeof FETCH_ALLPROJECTS_REQUEST;
+}
+
+export interface FetchProjectByIdRequest {
+  type: typeof FETCH_PROJECT_BYID_REQUEST;
+  params: number;
+}
+
+export type FetchProjectSuccess = {
```
Comment (sourcecoursepoc) [inline src/redux/actions/types.tsx:224]: Before committing code make sure eslint issues are solved and formatting is done

File: src/redux/sagas/projectSaga.tsx [hunk 13]
```diff
@@ -1,22 +1,23 @@
...
-import { fetchProjectFailure, fetchProjectSuccess } from "../actions/fetchProjectAction";
+import { IPROJECT, DeleteProjectInfoAction, DeleteProjectGroupsInfoAction } from "../actions/types";
+import { FETCH_ALLPROJECTS_REQUEST } from "../actions/actionTypes";
+import { fetchProjectFailure, fetchProjectSuccess, deleteProjectInfoActionSuccess, deleteProjectInfoActionFailure, deleteProjectGroupsInfoAction, deleteProjectGroupsInfoActionSuccess, deleteProjectGroupsInfoActionFailure } from "../actions/fetchProjectAction";
+import { DELETE_PROJECTS_INFO_ACTION } from "../actions/projectActionTypes";
 
 
 const getProjects = () =>
-axios.get<IPROJECT[]>("http://localhost:8080/sourcecourse/home");
+  axios.get<IPROJECT[]>("http://localhost:8080/sourcecourse/home");
```
Comment (sourcecoursepoc) [inline src/redux/sagas/projectSaga.tsx]: Are we keeping the base urls in individual files or in a constant

File: src/redux/sagas/projectSaga.tsx [hunk 14]
```diff
@@ -31,12 +32,52 @@ function* fetchProjectSaga():any {
   }
 }
 
+
+// delete project saga
+
+// const URL = 'http://localhost:8080/sourcecourse/project';
+
+// function* deleteProject(action: DeleteProjectInfoAction) {
+//   try {
```
Comment (sourcecoursepoc) [inline src/redux/sagas/projectSaga.tsx]: Remove

**Review by sourcecoursepoc — CHANGES_REQUESTED** [review]

File: src/components/ComposePage/TextArea/textArea.tsx (hunk 1 above)
Comment (211458-LakkeRajashekar) [reply src/components/ComposePage/TextArea/textArea.tsx]: done

File: src/pages/compose/index.tsx (hunk 2 above)
Comment (211458-LakkeRajashekar) [reply src/pages/compose/index.tsx]: resolved

File: src/redux/actions/fetchProjectAction.tsx (hunk 3 above)
Comment (211458-LakkeRajashekar) [reply src/redux/actions/fetchProjectAction.tsx]: resolved

File: src/components/homeLeftArea/homeLeftArea.tsx (hunk 7 above)
Comment (211458-LakkeRajashekar) [reply src/components/homeLeftArea/homeLeftArea.tsx:17]: while doing prettier code format the space is coming automatically

File: src/components/homeLeftArea/homeLeftArea.tsx (hunk 8 above)
Comment (211458-LakkeRajashekar) [reply src/components/homeLeftArea/homeLeftArea.tsx:50]: while doing prettier code format the space is coming automatically

File: src/redux/sagas/projectSaga.tsx (hunk 13 above)
Comment (211458-LakkeRajashekar) [reply src/redux/sagas/projectSaga.tsx]: we are not keeping base url in individual file

**Review by sourcecoursepoc — APPROVED** [review]

**Comment by sourcecoursepoc** [comment]
Not relevant anymore

**Pull request closed without merging**